tests/func/experiments/test_remove.py

```diff
@@ -0,0 +1,49 @@
+from funcy import first
+
+from dvc.repo.experiments.utils import exp_refs_by_rev
+
+
+def test_remove_experiments_by_ref(tmp_dir, scm, dvc, exp_stage, caplog):
+    results = dvc.experiments.run(exp_stage.addressing, params=["foo=2"])
+    exp = first(results)
+    ref_info = first(exp_refs_by_rev(scm, exp))
+
+    assert dvc.experiments.remove(["non-exist"]) == 0
+    assert ("'non-exist' is not a valid experiment") in caplog.text
+
+    removed = dvc.experiments.remove([str(ref_info)])
+    assert removed == 1
+    assert scm.get_ref(str(ref_info)) is None
+
+
+def test_remove_all_queued_experiments(tmp_dir, scm, dvc, exp_stage):
+    queue_length = 3
+    for i in range(queue_length):
+        dvc.experiments.run(
+            exp_stage.addressing, params=[f"foo={i}"], queue=True
+        )
+
+    assert len(dvc.experiments.stash) == queue_length
+    removed = dvc.experiments.remove(queue=True)
+    assert removed == queue_length
+    assert len(dvc.experiments.stash) == 0
+
+
+def test_remove_special_queued_experiments(tmp_dir, scm, dvc, exp_stage):
```

Comment:
More test case ideas to consider:
* Assert that committed exp is not removed here?
* Assert that multiple queued exp refs can be removed at once?
* Remove a mix of queued exps and committed exp refs?